src/contracts/core/OperatorSetManager.sol

```diff
@@ -0,0 +1,208 @@
+                return;
+            } else if (_lockedMagnitudeUpdates[operator][strategy][lockedMagnitudeUpdatesLength - 1].endEpoch + 1 == epoch) {
+                // extend the last lock to this epoch
+                _lockedMagnitudeUpdates[operator][strategy][lockedMagnitudeUpdatesLength - 1].endEpoch = epoch;
+                // emit MagnitudeUpdatesLocked(operator, strategy, epoch);
+                return;
+            }
+        }
+
+        _lockedMagnitudeUpdates[operator][strategy].push(StakeLock({startEpoch: epoch, endEpoch: epoch}));
+
+        // emit MagnitudeUpdatesLocked(operator, strategy, epoch);
+    }
+
+    /// VIEW
+
+    /**
+     * @param operator the operator to get the slashable bips for
+     * @param operatorSet the operator set to get the slashable bips for
+     * @param strategy the strategy to get the slashable bips for
+     * @param epoch the epoch to get the slashable bips for for
+     *
+     * @return slashableBips the slashable bips of the given strategy owned by
+     * the given OperatorSet for the given operator and epoch
+     */
+    function getSlashableBips(
+        address operator,
+        OperatorSet calldata operatorSet,
+        IStrategy strategy,
+        uint32 epoch
+    ) public view returns (uint16 slashableBips) {
+        require(epoch <= EpochUtils.getNextSlashingParameterEffectEpoch(), "Epoch is more than 2 epochs in the future");
+
+        // if locked for this epoch, set the epoch to the start of the lock
+        for (uint256 i = _lockedMagnitudeUpdates[operator][strategy].length; i > 0; i--) {
+            if (_lockedMagnitudeUpdates[operator][strategy][i - 1].startEpoch <= epoch && _lockedMagnitudeUpdates[operator][strategy][i - 1].endEpoch >= epoch) {
+                epoch = _lockedMagnitudeUpdates[operator][strategy][i - 1].startEpoch;
+                break;
+            }
+        }
+
+        MagnitudeUpdate[] storage operatorSetMagnitudeUpdates = _operatorSetMagnitudeUpdates[operator][strategy][_hashOperatorSet(operatorSet)];
+        uint192 operatorSetMagnitude = 0;
+
+        // iterate from the latest operator set magnitude update to the oldest
+        for (uint256 i = operatorSetMagnitudeUpdates.length; i > 0; i--) {
+            if (operatorSetMagnitudeUpdates[i - 1].epoch <= epoch) {
+                operatorSetMagnitude = operatorSetMagnitudeUpdates[i - 1].magnitude;
+                break;
+            }
+        }
+
+        uint192 totalMagnitude = 0;
+        // iterate through total magnitude updates to get the total magnitude
+        for (uint256 i = _totalMagnitudeUpdates[operator][strategy].length; i > 0; i--) {
+            if (_totalMagnitudeUpdates[operator][strategy][i - 1].epoch <= epoch) {
+                totalMagnitude = _totalMagnitudeUpdates[operator][strategy][i - 1].totalMagnitude;
+                break;
+            }
+        }
+
+        return uint16(operatorSetMagnitude * SlashingAccountingUtils.BIPS_FACTOR / totalMagnitude);
```

Comment:
should be noted that if there are totalMagnitude updates then this view function will revert 
since `totalMagnitude = 0`